src/components/DeploymentFormV2.vue

```diff
@@ -0,0 +1,165 @@
+
+    <p-checkbox v-model="shouldValidate" label="Validate parameters before submitting" />
+
+    <p-label label="Enforce Parameter Schema">
+      <p-toggle v-model="enforceParameterSchema" :disabled="!parameters" />
+    </p-label>
+
+    <p-divider />
+
+    <h3 class="deployment-form__heading">
+      {{ localization.info.infraOverrides }}
+    </h3>
+    <p-label label="Infrastructure Overrides (Optional)" :message="overrideError" :state="overrideState">
+      <p-code-input
+        v-model="infrastructureOverrides"
+        show-line-numbers
+        :min-lines="3"
+        class="deployment-form__json"
+        show-format-button
+      />
+    </p-label>
+
+    <template #footer>
+      <p-button @click="cancel">
+        Cancel
+      </p-button>
+      <p-button primary type="submit" @click="submit">
+        Save
+      </p-button>
+    </template>
+  </p-form>
+</template>
+
+<script lang="ts" setup>
+  import { showToast } from '@prefecthq/prefect-design'
+  import { useValidation, useValidationObserver } from '@prefecthq/vue-compositions'
+  import { computed, h, ref } from 'vue'
+  import { WorkPoolCombobox, WorkPoolQueueCombobox } from '@/components'
+  import ToastParameterValidationError from '@/components/ToastParameterValidationError.vue'
+  import { localization } from '@/localization'
+  import { Deployment, DeploymentUpdateV2 } from '@/models'
+  import { SchemaInputV2 } from '@/schemas'
+  import { useSchemaValidation } from '@/schemas/compositions/useSchemaValidation'
+  import { stringify, isJson } from '@/utilities'
+
+  const props = defineProps<{
+    deployment: Deployment,
+  }>()
+
+  const name = computed(() => props.deployment.name)
+  const description = ref(props.deployment.description)
+  const workPoolName = ref(props.deployment.workPoolName)
+  const workQueueName = ref(props.deployment.workQueueName)
+  const parameters = ref(props.deployment.parametersV2)
+  const tags = ref(props.deployment.tags)
+  const infrastructureOverrides = ref(stringify(props.deployment.infrastructureOverrides))
+  const enforceParameterSchema = ref(props.deployment.enforceParameterSchema)
+  const shouldValidate = ref(true)
+
+  const schema = computed(() => {
+    return { ...props.deployment.parameterOpenApiSchemaV2, required: [] }
+  })
+
+  const { validate } = useValidationObserver()
+  const { errors, validate: validateParameters } = useSchemaValidation(schema, parameters)
+  const { state: overrideState, error: overrideError } = useValidation(infrastructureOverrides, isJson('Infrastructure overrides'))
+
+  const emit = defineEmits<{
+    (event: 'submit', value: DeploymentUpdateV2): void,
+    (event: 'cancel'): void,
+  }>()
+
+  async function submit(): Promise<void> {
+    if (shouldValidate.value) {
+      try {
+        const valid = (await Promise.all([
+          validate(),
+          validateParameters(),
```

Comment:
I don't think so, because on line 100 we're modifying the schema to remove all required properties. Making everything optional. And then we're sending that modified schema to the validation endpoint.

Reply:
i was looking for that code and still skipped over it 😮‍💨